mountpoint-s3/src/inode.rs

```diff
@@ -531,7 +532,11 @@ impl Superblock {
                             error=?e,
                             "DeleteObject failed for unlink",
                         );
-                        Err(InodeError::ClientError(anyhow!(e).context("DeleteObject failed")))?;
+                        let metadata = e.meta().clone();
+                        Err(InodeError::ClientError {
+                            source: anyhow!(e).context("DeleteObject failed"),
+                            metadata,
+                        })?;
```

Comment:
This calls for a "constructor" on `InodeError` that takes a `ProvideErrorMetadata` source and a context.